### PR TITLE
COMP: Simplify static array length checking

### PR DESCRIPTION
**Pull request by hjmjohnson**
Use c++17 features to enforce compile-time
sanity checks on array lengths.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Review by thewtex — APPROVED** [review]
:sunglasses:

**Review by N-Dekker — APPROVED** [review]
Strictly speaking, you'd need to `#include <iterator>` in order to use `std::size`. But apparently it is already `#include`'d indirectly.

Approved 👍  I leave it up to you if you still want to do `#include <iterator>`

**Pull request merged**
